### PR TITLE
[PHP] Fix some PHP 7.2 deprecations

### PR DESCRIPTION
**Pull request by andrerom**
Todo:
- [x] Testing
- [x] Review

### CODE REVIEW

File: lib/eztemplate/classes/eztemplatecompiler.php [hunk 1]
```diff
@@ -1197,7 +1197,7 @@ static function processElementTransformationList( $tpl, &$node, $elements, &$pri
     static function processElementTransformationChild( $useComments, $php, $tpl, &$node,
                                                 $elementTree, $elementList, &$resourceData )
     {
-        $count = count( $elementList );
+        $count = $elementList instanceof Countable || is_array( $elementList ) ? count( $elementList ) : 0;
```
Comment (andrerom) [inline lib/eztemplate/classes/eztemplatecompiler.php]: According to doc count() returns 1 on non countable / arrays, but 0 on null. As variable is used as array beneath this is assuming 0 if not countable / array.

Comment (emodric) [reply lib/eztemplate/classes/eztemplatecompiler.php]: Can we use https://github.com/symfony/polyfill-php73 and its `is_countable` method for this?

File: kernel/private/classes/clusterfilehandlers/ezdfsfilehandler.php [hunk 2]
```diff
@@ -153,9 +153,21 @@ public function loadMetaData( $force = false )
...
-                                    '$a=$a["cnt"]; $b=$b["cnt"]; if ( $a > $b ) return -1; else if ( $a == $b ) return 0; else return 1;' ) );
+            usort( $GLOBALS['eZClusterInfo'], function ( $a, $b ) {
+                $a = $a['cnt'];
+                $b = $b['cnt'];
+
+                if ( $a > $b )
+                {
+                    return -1;
+                }
+                else if ( $a == $b )
```
Comment (emodric) [inline kernel/private/classes/clusterfilehandlers/ezdfsfilehandler.php]: No need for `else`

File: lib/ezutils/classes/ezphpcreator.php [hunk 3]
```diff
@@ -565,7 +565,7 @@ function thisVariableText( $value, $column = 0, $iteration = 0, $maxIterations =
                 $column += strlen( $text );
                 $valueKeys = array_keys( $value );
                 $isIndexed = true;
-                for ( $i = 0; $i < count( $valueKeys ); ++$i )
+                for ($i = 0, $count = count($valueKeys); $i < $count; ++$i )
```
Comment (glye) [inline lib/ezutils/classes/ezphpcreator.php]: FWIW, unbalanced use of space-inside-parentheses.

File: lib/ezutils/classes/ezphpcreator.php [hunk 4]
```diff
@@ -676,7 +676,7 @@ static function variableText( $value, $column = 0, $iteration = 0, $maxIteration
             $column += strlen( $text );
             $valueKeys = array_keys( $value );
             $isIndexed = true;
-            for ( $i = 0; $i < count( $valueKeys ); ++$i )
+            for ($i = 0, $count = count($valueKeys); $i < $count; ++$i )
```
Comment (glye) [inline lib/ezutils/classes/ezphpcreator.php]: Unbalanced use of space-inside-parentheses.

**Pull request merged**